### ISSUE TITLE
libFuzzer sometimes doesn't report timeout

### ISSUE THREAD

**Issue by adombeck** [llvm/llvm-project#53429]
This fuzz target should always trigger a timeout:

```
cat << EOF > trigger_timeout.c
#include <stddef.h>
#include <stdint.h>
#include <unistd.h>

int LLVMFuzzerTestOneInput(const uint8_t *Data, size_t Size) {
  sleep(10000);
  return 0;
}
EOF
clang -fsanitize=fuzzer trigger_timeout.c
```

... and most of the time it does:

```
INFO: Running with entropic power schedule (0xFF, 100).
INFO: Seed: 1
INFO: Loaded 1 modules   (1 inline 8-bit counters): 1 [0x482fa0, 0x482fa1), 
INFO: Loaded 1 PC tables (1 PCs): 1 [0x470150,0x470160), 
INFO: -max_len is not provided; libFuzzer will not generate inputs larger than 4096 bytes
ALARM: working on the last Unit for 1 seconds
       and the timeout value is 1 (use -timeout=N to change)
MS: 0 ; base unit: 0000000000000000000000000000000000000000


artifact_prefix='./'; Test unit written to ./timeout-da39a3ee5e6b4b0d3255bfef95601890afd80709
Base64: 
==1835077== ERROR: libFuzzer: timeout after 1 seconds
[...]
```

... but sometimes it just passes after 1 or 2 seconds:

```
while ! ./a.out -timeout=1 -runs=1 -seed=1; do done
[...]
INFO: Running with entropic power schedule (0xFF, 100).
INFO: Seed: 1
INFO: Loaded 1 modules   (1 inline 8-bit counters): 1 [0x482fa0, 0x482fa1), 
INFO: Loaded 1 PC tables (1 PCs): 1 [0x470150,0x470160), 
INFO: -max_len is not provided; libFuzzer will not generate inputs larger than 4096 bytes
INFO: A corpus is not provided, starting from an empty corpus
#2	pulse  ft: 1 exec/s: 1 rss: 26Mb
#2	INITED cov: 1 ft: 1 corp: 1/1b exec/s: 1 rss: 26Mb
#2	DONE   cov: 1 ft: 1 corp: 1/1b lim: 4 exec/s: 1 rss: 26Mb
Done 2 runs in 2 second(s)
```

My clang version:
```
clang --version
Debian clang version 13.0.1-+rc3-1~exp1
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin
```

**Comment by adombeck** [comment]
I can't reproduce the issue with this fuzz target:

```C
#include <stddef.h>
#include <stdint.h>

int LLVMFuzzerTestOneInput(const uint8_t *Data, size_t Size) {
  while(1);
  return 0;
}
```
